python/test_gilded_rose.py

```diff
@@ -5,12 +5,153 @@
+		self.assertEqual("Aged Brie", items[0].name)
+		self.assertEqual(99, items[0].sell_in)
+		self.assertEqual(31, items[0].quality)
+		self.assertEqual("Sulfuras, Hand of Ragnaros", items[1].name)
+		self.assertEqual(30, items[1].sell_in)
+		self.assertEqual(80, items[1].quality)
+		self.assertEqual("Chips", items[2].name)
+		self.assertEqual(4, items[2].sell_in)
+		self.assertEqual(9, items[2].quality)
+
+	def test_update_quality_multiple(self):
+		items = [Item("Aged Brie", 100, 30),
+				 Item("Sulfuras, Hand of Ragnaros", 30, 80), Item("Chips", 5, 10)]
+		gilded_rose = GildedRose(items)
+		for i in range(10):
+			gilded_rose.update_quality()
+		self.assertEqual("Aged Brie", items[0].name)
+		self.assertEqual(90, items[0].sell_in)
+		self.assertEqual(40, items[0].quality)
+		self.assertEqual("Sulfuras, Hand of Ragnaros", items[1].name)
+		self.assertEqual(30, items[1].sell_in)
+		self.assertEqual(80, items[1].quality)
+		self.assertEqual("Chips", items[2].name)
+		self.assertEqual(-5, items[2].sell_in)
+		self.assertEqual(0, items[2].quality)
+
+	def test_quality_upper_limit(self):
+		items = [Item("Aged Brie", 100, 30),
+				 Item("Backstage passes to a TAFKAL80ETC concert", 100, 10)]
+		gilded_rose = GildedRose(items)
+		for i in range(50):
+			gilded_rose.update_quality()
+		self.assertEqual("Aged Brie", items[0].name)
+		self.assertEqual(50, items[0].sell_in)
+		self.assertEqual(50, items[0].quality)
+		self.assertEqual("Backstage passes to a TAFKAL80ETC concert", items[1].name)
+		self.assertEqual(50, items[1].sell_in)
+		self.assertEqual(50, items[1].quality)
+
+	def test_quality_lower_limit(self):
+		items = [Item("Elixir of the Mongoose", 5, 10)]
+		gilded_rose = GildedRose(items)
+		for i in range(50):
+			gilded_rose.update_quality()
+		self.assertEqual("Elixir of the Mongoose", items[0].name)
+		self.assertEqual(-45, items[0].sell_in)
+		self.assertEqual(0, items[0].quality)
+
+	def test_quality_expired(self):
+		items = [Item("Elixir of the Mongoose", 5, 50)]
+		gilded_rose = GildedRose(items)
+		for i in range(15):
+			gilded_rose.update_quality()
+		self.assertEqual("Elixir of the Mongoose", items[0].name)
+		self.assertEqual(-10, items[0].sell_in)
+		self.assertEqual(25, items[0].quality)
+
+	def test_backstage_pass(self):
+		items = [Item("Backstage passes to a TAFKAL80ETC concert", 11, 10),
+				 Item("Backstage passes to a Eminem concert", 11, 10),
+				 Item("Backstage passes to a Kendrick Lamar concert", 11, 10),]
+		gilded_rose = GildedRose(items)
+		for i in range(10):
+			gilded_rose.update_quality()
+		self.assertEqual("Backstage passes to a TAFKAL80ETC concert", items[0].name)
+		self.assertEqual(1, items[0].sell_in)
+		self.assertEqual(33, items[0].quality)
+		self.assertEqual("Backstage passes to a Eminem concert", items[1].name)
+		self.assertEqual(1, items[1].sell_in)
+		self.assertEqual(33, items[1].quality)
+		self.assertEqual("Backstage passes to a Kendrick Lamar concert", items[2].name)
```

Comment:
Love for K-dot 🎶

Reply:
Prince of rap. Big 3 ❌, Big Me (Kendrick) ✅